samples/boards/nrf/system_off/src/retained_mem.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "retained.h"
+
+#include <stdint.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/retained_mem.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/crc.h>
+
+const static struct device *retained_mem_device = DEVICE_DT_GET(DT_ALIAS(retainedmemdevice));
+
+struct retained_data retained;
```

Comment:
Is there a reason for this variable to be in the global scope?

Reply:
yes - this structure is referenced directly by sample `main.c` code: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/nrf/system_off/src/main.c#L40-L43 and this is a scheme enforced by existing POWER-based implementation for nRF52: https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/boards/nrf/system_off/src/retained.c

I would like to avoid major refactors of this sample, as the goal was to add support for nRF54L15 rather than making it "perfect"